docs: add security and privacy considerations (#260)
allinbits/cosmos-cash: docs/Explanation/ADR/adr-004-did.md

MsgCreateDidDocument(id string, signerPubKey string)
```

> The message parameters are the DID to be created and the `signerPubKey`. The `signerPubKey` MUST be the public key of the account that signs the transaction. The public key MUST be used to attach a verification method of type `EcdsaSecp256k1VerificationKey2019` with the value of `publicKeyMultibase` that contains the public key encoded according to the [hexadecimal multibase format](https://datatracker.ietf.org/doc/html/draft-multiformats-multibase#appendix-B.1).

Reviewer: ```suggestion
The message parameters are the DID to be created and the `signerPubKey`. The `signerPubKey` MUST be the public key of the account that signs the transaction. The public key MUST be used to attach a verification method of type `EcdsaSecp256k1VerificationKey2019` with the value of `publicKeyMultibase` that contains the public key encoded according to the Multibase Data Format [hexadecimal Hexadecimal upper-case encoding](https://datatracker.ietf.org/doc/html/draft-multiformats-multibase#appendix-B.1).
```

Reviewer: oh, for got to save this suggestion on first review... the goal is to make link text descriptive, so using the title of the section for the destination page is a better signpost for our readers

The format for the queries is:
-->

## Privacy Considerations

> When any data (e.g. W3C Verifiable Credentials) is associated with Cosmos DIDs, sharing that data would also impose sharing the onchain data graph (e.g. transaction history) of the blockchain account that controls the DID.
> 
> Using personal identifiable information as DID Method specific identifiers (e.g. alice) discloses personal information every time the DID is shared with a counter party. This specification DOES NOT endorse the use of identifiers that correlates to human beings or other sensible subjects.

Reviewer: ```suggestion
When any data (for example, W3C Verifiable Credentials) is associated with Cosmos DIDs, sharing that data would also impose sharing the on-chain data graph (for example, transaction history) of the blockchain account that controls the DID.

Using personally identifiable information as DID Method-specific identifiers (for example, account name alice) discloses personal information every time the DID is shared with a counterparty. This specification DOES NOT endorse the use of identifiers that correlates to human beings or other sensible subjects.
```

Reviewer: let's avoid Latin and spell out "for example" 
I think "alice" is an example of account name alice? or where else would alice be disclosed?

## Security Considerations

> Ephemeral DIDs (`did:cosmos:key` type) are generated based on a blockchain address, if access to the authoritative keys for an account are lost, the control of the DID and verifiable data issued by the DID is lost as well. 

Reviewer: ```suggestion
Ephemeral DIDs (`did:cosmos:key` type) are generated based on a blockchain address. If access to the authoritative keys for an account are lost, the control of the DID and verifiable data issued by the DID is lost as well. 
```
